### PR TITLE
NODE-2608 Removed Diff from tests

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: node-it/src/test/scala/com/wavesplatform/it/async/WideStateGenerationSuite.scala [hunk 1]
```diff
@@ -86,9 +85,9 @@ class WideStateGenerationSuite extends BaseFreeSpec with WaitForHeight2 with Tra
       blocks <- node.blockSeq(1, height)
     } yield {
       val txsInBlockchain = blocks.flatMap(_.transactions.map(_.id))
-      val diff            = txIds -- txsInBlockchain
+      val snapshot        = txIds -- txsInBlockchain
       withClue(s"all transactions in node") {
-        diff shouldBe empty
+        snapshot shouldBe empty
```
Comment (vsuharnikov) [inline node-it/src/test/scala/com/wavesplatform/it/async/WideStateGenerationSuite.scala]: Same

File: node/src/test/scala/com/wavesplatform/state/diffs/AssetTransactionsDiffTest.scala [hunk 2]
```diff
@@ -42,23 +43,34 @@ class AssetTransactionsDiffTest extends PropSpec with BlocksTransactionsHelpers
...
-      newState.balance(issue.sender.toAddress, reissue.asset) shouldEqual totalAssetVolume
+    withDomain(RideV3) { d =>
+      d.appendBlock(gen)
+      d.appendBlock(issue)
+      d.appendBlock(reissue, burn)
+      val assetQuantityDiff = reissue.quantity.value - burn.quantity.value
+      d.liquidSnapshot.balances.toSeq
+        .map {
+          case ((`defaultAddress`, Waves), amount) =>
+            Waves -> (amount - d.rocksDBWriter.balance(defaultAddress, Waves) + (-issue.fee.value + reissue.fee.value + burn.fee.value) / 5 * 3)
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/AssetTransactionsDiffTest.scala]: Where did 3/5 come from? Could you introduce a variable with a name or just add a comment? Is it a carry fee coefficient? 
May be we should just append an empty block to eliminate such calculations and make whole `collect` (`map` in other places) block simpler? Such fee calculations are not essential for this test.

File: node-it/src/test/scala/com/wavesplatform/it/async/MicroblocksGenerationSuite.scala [hunk 3]
```diff
@@ -24,8 +24,8 @@ class MicroblocksGenerationSuite extends BaseFreeSpec with TransferSending {
       block.transactions.size shouldBe maxTxs
 
       val blockTxs = block.transactions.map(_.id)
-      val diff     = uploadedTxs.map(_.id).toSet -- blockTxs
-      diff shouldBe empty
+      val snapshot = uploadedTxs.map(_.id).toSet -- blockTxs
+      snapshot shouldBe empty
```
[the remark this answers is not in the document]
Comment (xrtm000) [reply node-it/src/test/scala/com/wavesplatform/it/async/MicroblocksGenerationSuite.scala]: Fixed

File: node-it/src/test/scala/com/wavesplatform/it/async/WideStateGenerationSuite.scala (hunk 1 above)
Comment (xrtm000) [reply node-it/src/test/scala/com/wavesplatform/it/async/WideStateGenerationSuite.scala]: Fixed

File: node/src/test/scala/com/wavesplatform/state/diffs/DataTransactionDiffTest.scala [hunk 4]
```diff
@@ -41,34 +43,35 @@ class DataTransactionDiffTest extends PropSpec with WithState {
...
-        state.balance(sender.toAddress) shouldBe (ENOUGH_AMT - txs.map(_.fee.value).sum)
-        state.accountData(sender.toAddress, item1.key) shouldBe Some(item3)
-        state.accountData(sender.toAddress, item2.key) shouldBe Some(item2)
+    withDomain(RideV3) { d =>
+      d.appendBlock(genesisTx)
+      d.appendBlock(txs(0), txs(1), txs(2))
+      d.liquidSnapshot.balances((sender, Waves)) shouldBe (ENOUGH_AMT - txs.map(_.fee.value).sum)
+      d.liquidSnapshot.accountData(sender)(item1.key) shouldBe item3
+      d.liquidSnapshot.accountData(sender)(item2.key) shouldBe item2
+      assertBalanceInvariant(d.liquidSnapshot, d.rocksDBWriter,  -(txs(0).fee.value + txs(1).fee.value + txs(2).fee.value) * 3 / 5)
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/DataTransactionDiffTest.scala]: Same question about a carry fee here

File: node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala [hunk 5]
```diff
@@ -93,29 +92,26 @@ class EthereumTransactionDiffTest extends FlatSpec with WithDomain with DiffMatc
...
+    withDomain(RideV6, Seq(AddrWithBalance(senderKp.toWavesAddress), AddrWithBalance(issuer.toAddress))) { d =>
       val issue          = TxHelpers.issue(issuer)
       val nativeTransfer = TxHelpers.transfer(issuer, senderKp.toWavesAddress, issue.quantity.value, issue.asset)
       val assetTransfer  = EthTxGenerator.generateEthTransfer(senderKp, recipient, issue.quantity.value, issue.asset, fee)
 
       d.appendBlock(issue, nativeTransfer)
-      assertBalanceInvariant(d.createDiff(assetTransfer).combineF(feeDiff).explicitGet())
-
       d.appendAndAssertSucceed(assetTransfer)
+      assertBalanceInvariant(d.liquidSnapshot, d.rocksDBWriter, 6.waves + (issue.fee.value + nativeTransfer.fee.value - assetTransfer.fee) * 3 / 5)
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala]: too

File: node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala [hunk 6]
```diff
@@ -93,29 +92,26 @@ class EthereumTransactionDiffTest extends FlatSpec with WithDomain with DiffMatc
...
-      val wavesTransfer = EthTxGenerator.generateEthTransfer(senderKp, recipient, 1.waves, Waves, fee)
-      assertBalanceInvariant(d.createDiff(wavesTransfer).combineF(feeDiff).explicitGet())
+    val fee = TestValues.fee
 
+    withDomain(RideV6.copy(rewardsSettings = RewardsVotingSettings(None)), Seq(AddrWithBalance(senderKp.toWavesAddress))) { d =>
+      val wavesTransfer   = EthTxGenerator.generateEthTransfer(senderKp, recipient, 1.waves, Waves, fee)
       val transferPayload = wavesTransfer.payload.asInstanceOf[Transfer]
 
       d.appendAndAssertSucceed(wavesTransfer)
+      assertBalanceInvariant(d.liquidSnapshot, d.rocksDBWriter, 6.waves - wavesTransfer.fee * 3 / 5)
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala]: too

File: node/src/test/scala/com/wavesplatform/state/diffs/ExchangeTransactionDiffTest.scala [hunk 7]
```diff
@@ -328,23 +329,26 @@ class ExchangeTransactionDiffTest extends PropSpec with Inside with WithDomain w
...
-
-        blockDiff.portfolios(exchange.sender.toAddress).balance shouldBe exchange.buyMatcherFee + exchange.sellMatcherFee - exchange.fee.value
+      withDomain(ScriptsAndSponsorship) { d =>
+        d.appendBlock(genesis*)
+        d.appendBlock(issue1, issue2)
+        d.appendBlock(exchange)
+        d.liquidSnapshot.balances.toSeq
+          .map {
+            case ((`defaultAddress`, Waves), amount) =>
+              Waves -> (amount - d.rocksDBWriter.balance(defaultAddress, Waves) - (issue1.fee.value + issue2.fee.value - exchange.fee.value) * 3 / 5)
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/ExchangeTransactionDiffTest.scala]: I think we can just search files with "3 / 5" substring.

File: node/src/test/scala/com/wavesplatform/state/diffs/LeaseTransactionsDiffTest.scala [hunk 8]
```diff
@@ -38,20 +39,20 @@ class LeaseTransactionsDiffTest extends PropSpec with WithDomain {
...
-        totalPortfolioDiff.balance shouldBe 0
-        total(totalPortfolioDiff.lease) shouldBe 0
-        totalPortfolioDiff.effectiveBalance(false).explicitGet() shouldBe 0
-        totalPortfolioDiff.assets.values.foreach(_ shouldBe 0)
+      assertDiffAndState(Seq(TestBlock.create(Seq(genesis, lease))), TestBlock.create(Seq(leaseCancel))) { case (snapshot, _) =>
+//        val totalPortfolioDiff = snapshot.portfolios.values.fold(Portfolio())(_.combine(_).explicitGet())
+//        totalPortfolioDiff.balance shouldBe 0
+//        total(totalPortfolioDiff.lease) shouldBe 0
+//        totalPortfolioDiff.effectiveBalance(false).explicitGet() shouldBe 0
+//        totalPortfolioDiff.assets.values.foreach(_ shouldBe 0)
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/state/diffs/LeaseTransactionsDiffTest.scala]: ?

File: node/src/test/scala/com/wavesplatform/state/diffs/AssetTransactionsDiffTest.scala (hunk 2 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/state/diffs/AssetTransactionsDiffTest.scala]: If we append an empty block `liquidSnapshot` will be cleared

File: node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala (hunk 6 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala]: Done

File: node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala (hunk 5 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/state/diffs/EthereumTransactionDiffTest.scala]: Done

File: node/src/test/scala/com/wavesplatform/state/diffs/DataTransactionDiffTest.scala (hunk 4 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/state/diffs/DataTransactionDiffTest.scala]: Done

File: node/src/test/scala/com/wavesplatform/state/diffs/ExchangeTransactionDiffTest.scala (hunk 7 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/state/diffs/ExchangeTransactionDiffTest.scala]: Done

File: node/src/test/scala/com/wavesplatform/state/diffs/LeaseTransactionsDiffTest.scala (hunk 8 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/state/diffs/LeaseTransactionsDiffTest.scala]: Fixed

**Review by vsuharnikov — APPROVED** [review]

**Review by ivan-mashonskiy — APPROVED** [review]

**Pull request merged**